### PR TITLE
Fixed error "Can't sign: unable to write 'random state'" if sending emails signed with S/MIME. – OTRS bug #14522

### PR DESCRIPTION
**Pull request by KaiHerlemann**
I solved the OTRS bug [#14522](https://bugs.otrs.org/show_bug.cgi?id=14522) this way. Because of copyright reasons, I don't copy the text of the OTRS bug.
In short: sending emails signed with S/MIME doesn't work, the email isn't signed. You get an error message in the system log "Can't sign: unable to write 'random state'". This is due to the file permissions, the group has no r/w permissions for the file `$TEMPDIR/.rnd`. If you set them automatically, the system will undo this. Therefore I used a variable so that each user (otrs/apache etc.) has its own .rnd file.
I think it's not working on Windows, probably it works on *BSD systems. I'm not sure how to solve it (maybe `my $username = getpwuid($<) || Win32::LoginName();`?), you have probably more knowledge about that. But locally on CentOS 7 it works.

It's just a workaround, actually I prefer real solutions, but I don't see any way how to solve it.

## Type of change

1 - 🐞 bug 🐞            - Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [ ] You improved or added new unit tests.(❕)
- [ ] Local ZnunyCodePolicy run passes successfully.(❕)
- [ ] Local unit tests pass.(❕)
- [x] GitHub workflow ZnunyCodePolicy passes.(❗)
- [x] GitHub workflow unit tests pass.(❗)

### CODE REVIEW

**Comment by dennykorsukewitz** [comment]
Hi @KaiHerlemann,

this PR is still in 'Draft' mode.
We only look at PRs when they are no longer marked as draft.
Please check.

Thx 🚀

**Comment by dennykorsukewitz** [comment]
Hi @KaiHerlemann 

```
ZnunyCodePolicy:

[checked] Kernel/System/Crypt/SMIME.pm
TidyAll::Plugin::OTRS::Perl::PerlCritic
Variable, subroutine, and package names have to be in CamelCase: $linux_username at line 2328, column 8. (no explanation).
```

Can you please fix this.
THX 🚀

**Comment by rkaldung** [comment]
I'm with this solution. The only change I suggest is changing https://github.com/znuny/Znuny/pull/72/files#diff-aafcd3eab2fe2fdba7915b9166fd5ec11b6992b4e3bda4486d34e1c3a0e9463aR2328 to `my $SystemUsername = $<;`

The reason is simple: I'd like to prevent a username lookup on systems with NIS, LDAP, etc.

File: Kernel/System/Crypt/SMIME.pm [hunk 1]
```diff
@@ -2324,8 +2324,10 @@ sub _Init {
         $Self->{Cmd} = "LC_MESSAGES=POSIX $Self->{Bin}";
     }
 
+    # determine System Username to make sure each user has an own .rnd file
+    my $SystemUsername = getpwuid($<);
```
Comment (dennykorsukewitz) [inline Kernel/System/Crypt/SMIME.pm]:
```suggestion
    my $SystemUsername = $<;
```

**Review by dennykorsukewitz — CHANGES_REQUESTED** [review]

**Comment by KaiHerlemann** [comment]
works for me.. didn't want to request @dennykorsukewitz for a review, clicked wrong

**Review by dennykorsukewitz — APPROVED** [review]
Looks good to me.
What do you say? @rkaldung

**Review by rkaldung — APPROVED** [review]

**Pull request merged**